File: chroma-league/src/test/java/net/booone/chromaleague/hud/parts/resource/ShyvanaDragonFuryBarTest.java
package net.booone.chromaleague.hud.parts.resource;

import net.booone.chromaleague.IntSteps;
import net.booone.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class ShyvanaDragonFuryBarTest extends AbstractResourceTest {
    @Test
    void testShyvanaDragonPoolBar() {
        final IntSteps intSteps = new IntSteps(0, 100, 5);
        final ShyvanaDragonFuryBar shyvanaDragonFuryBar = new ShyvanaDragonFuryBar();
        new AnimationTester()
                .withBeforeIterationAction(i -> {
                    if (i >= 20 && i <= 50) {
                        mockResource(100, 100);
                    } else {
                        mockResource(intSteps.nextInt(), 100);
                    }
                })
                .testAnimation(shyvanaDragonFuryBar, 70);

    }
}

File: test/hud/parts/resource/ShyvanaDragonFuryBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {ShyvanaDragonFuryBar} from '../../../../src/hud/parts/resource/ShyvanaDragonFuryBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for Shyvana dragon fury resource bar on actual hardware. */
describe('ShyvanaDragonFury Integration', () => {
    setupResourceTest();

    test('testShyvanaDragonPoolBar', async () => {
        const intSteps = new IntSteps(0, 100, 5);
        const shyvanaDragonFuryBar = new ShyvanaDragonFuryBar();

        await new AnimationTester()
            .withBeforeIterationAction(i => {
                if (i >= 20 && i <= 50) {
                    mockResource(100, 100);
                } else {
                    mockResource(intSteps.nextInt(), 100);
                }
            })
            .testAnimation(shyvanaDragonFuryBar, 70);
    }, 30000);
});
